### ISSUE TITLE
condition_variable wait failed: Invalid argument (Crash on iOS)

### ISSUE THREAD

**Issue by mb12** [mapbox/mapbox-gl-native#1240]
I hit this crash today.  There is not much in the stack trace except for the following.

"
libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: condition_variable wait failed: Invalid argument
"
```
libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: condition_variable wait failed: Invalid argument

(lldb) bt
* thread #1: tid = 0x1459b2, 0x380eedfc libsystem_kernel.dylib`__pthread_kill + 8, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
  * frame #0: 0x380eedfc libsystem_kernel.dylib`__pthread_kill + 8
    frame #1: 0x3816cd36 libsystem_pthread.dylib`pthread_kill + 62
    frame #2: 0x3808e908 libsystem_c.dylib`abort + 76
    frame #3: 0x373ca9c8 libc++abi.dylib`abort_message + 88
    frame #4: 0x373e463a libc++abi.dylib`default_terminate_handler() + 214
    frame #5: 0x37aa8f48 libobjc.A.dylib`_objc_terminate() + 228
    frame #6: 0x373e1de2 libc++abi.dylib`std::__terminate(void (*)()) + 78
    frame #7: 0x373e18ae libc++abi.dylib`__cxa_rethrow + 102
    frame #8: 0x37aa8dd2 libobjc.A.dylib`objc_exception_rethrow + 42
    frame #9: 0x2a20045c CoreFoundation`CFRunLoopRunSpecific + 632
    frame #10: 0x2a2001d2 CoreFoundation`CFRunLoopRunInMode + 106
    frame #11: 0x315fe0a8 GraphicsServices`GSEventRunModal + 136
    frame #12: 0x2d80e7b0 UIKit`UIApplicationMain + 1440
    frame #13: 0x000f172e testapp`main(argc=1, argv=0x01172a5c) + 162 at main.m:15
```

**Comment by jfirebaugh** [comment]
Would be nice if it would print the stack trace for the original throw rather than for `__cxa_rethrow`...

Is there any way to replicate this?

**Comment by mb12** [comment]
I don't have a way to replicate this. I've seen this only once. I added an issue here for tracking purposes.

**Comment by 1ec5** [comment]
> Would be nice if it would print the stack trace for the original throw rather than for __cxa_rethrow...

Most likely we’d get a more informative stack trace if there had been a breakpoint on `-[NSException raise]` or `objc_exception_throw()`. (In Xcode, this is called an Objective-C exception breakpoint – always a good first thing to do when you move into a brand new workspace.)

**Comment by 1ec5** [comment]
There isn’t enough information to go on here. If you see the crash again, please open a new ticket, ideally with a stack trace from when the exception is raised.

**Comment by picciano** [comment]
FYI. I just saw this error today too. No additional information however.
